client/components/common/mixins/deprecation.js

```diff
@@ -0,0 +1,21 @@
+import { noCase } from 'change-case';
+
+const printDeprecationWarning = (oldEvent, newEvent) => {
+  console.warn(`Deprecation notice:
+    '${oldEvent}' listener is deprecated and will no longer be used!
+    Please emit '${newEvent}' instead on your content containers`);
+};
+
+export default {
+  methods: {
+    deprecateEvent(handler, { oldEvent, newEvent, adaptArgs }) {
+      newEvent = newEvent || noCase(oldEvent, { delimiter: ':' });
+      const context = this;
+      return function () {
```

Comment:
Because of `arguments` binding. Arrow functions does not have them (i.e. they access arguments of the first non-arrow parent) if I'm not wrong. Leveraging them here to be able to decorate methods with different signatures...
I'll check it one more time before I resolve this one...